src/wallet/wallet_tx_builder.cpp

```diff
@@ -680,9 +684,7 @@ TransactionBuilderResult TransactionEffects::ApproveAndBuild(
             [&](const SproutPaymentAddress& addr) {
                 builder.SendChangeToSprout(addr);
             },
-            [&](const RecipientAddress& addr) {
-                builder.SendChangeTo(addr, internalOVK);
-            }
+            [](const RecipientAddress&) { }
```

Comment:
I thought that there was, somewhere in here, an assertion that the total of payments was equal to the total value of inputs. Is that in the transaction builder itself?

Reply:
I didn’t see one in your original PR, or in the existing `TransactionBuilder`. Just above here, your code is summing up (most of) the spends, and dropping notes if it ever goes above the target, but my guess is that that’s from a point before you were thinking the change calculation would be handled in `WalletTxBuilder`. I fixed up that code and added an assert.